plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -906,12 +906,19 @@ public VersionNumber getLatestPluginVersion(Plugin dependendantPlugin, String pl
                 String pluginName = pluginInfo[0];
                 String pluginVersion = pluginInfo[1];
                 Plugin dependentPlugin = new Plugin(pluginName, pluginVersion, null, null);
+                dependentPlugin.setOptional(dependency.contains("resolution:=optional"));
+
                 if (useLatestSpecified && plugin.isLatest() || useLatestAll) {
-                    VersionNumber latestPluginVersion = getLatestPluginVersion(plugin, pluginName);
-                    dependentPlugin.setVersion(latestPluginVersion);
-                    dependentPlugin.setLatest(true);
+                    try {
+                        VersionNumber latestPluginVersion = getLatestPluginVersion(plugin, pluginName);
+                        dependentPlugin.setVersion(latestPluginVersion);
+                        dependentPlugin.setLatest(true);
+                    } catch(PluginNotFoundException e) {
+                        if (!dependentPlugin.getOptional()) {
```

Comment:
this isn't the only place where we're doing this check so seems ok,

you could also look at adjusting https://github.com/jenkinsci/plugin-installation-manager-tool/blob/8926b6e0449c0e6c146e86e8c314aa2d8fa261f0/plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java#L847 to not throw if it's optional